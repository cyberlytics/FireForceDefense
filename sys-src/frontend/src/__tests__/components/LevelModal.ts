import { mount, RouterLinkStub } from '@vue/test-utils';
import LevelModal from '@components/levelModal.vue';

describe('LevelModal.vue', () => {
    test('level modal matches snapshot', () => {
        const wrapper = mount(LevelModal, {
            propsData: {
                modalId: 'level-menu-modal',
            },
            mocks: {
                $t: (key: string) => key,
            },
            stubs: {
                RouterLink: RouterLinkStub,
            },
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});
