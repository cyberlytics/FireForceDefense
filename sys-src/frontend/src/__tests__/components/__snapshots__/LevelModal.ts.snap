// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LevelModal.vue level modal matches snapshot 1`] = `
<div
  class="modal"
  id="level-menu-modal"
>
  <div
    class="modal-dialog modal-dialog-centered"
  >
    <div
      class="modal-content"
    >
      <div
        class="modal-header"
      >
        <h2
          class="modal-title h5"
        >
          menu
        </h2>
         
        <button
          aria-label="Close"
          class="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
       
      <div
        class="modal-body"
      >
        <div
          class="text-center"
        >
          
                    Game is paused
                
        </div>
         
        <hr
          class="my-3"
        />
         
        <button
          class="btn btn-primary btn-block btn-lg"
        >
          
                    back to level selection
                
        </button>
         
        <!---->
         
        <hr
          class="my-3"
        />
         
        <button
          class="btn btn-primary btn-block btn-lg"
        >
          
                    Restart Level
                
        </button>
         
        <!---->
         
        <hr
          class="my-3"
        />
         
        <button
          aria-label="Close"
          class="btn btn-primary btn-block btn-lg"
          data-dismiss="modal"
        >
          
                    continue playing
                
        </button>
      </div>
    </div>
  </div>
</div>
`;
